### PR TITLE
linux_6_11: init at 6.11

### PR DESCRIPTION
**Pull request by K900**
## Description of changes

LWN summary: https://lwn.net/Articles/982034/ and https://lwn.net/Articles/982605/

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by alyssais — APPROVED** [review]

File: pkgs/os-specific/linux/kernel/perf/default.nix [hunk 1]
```diff
@@ -63,7 +63,7 @@ stdenv.mkDerivation {
 
   # Fix 6.10.0 holding pkg-config completely wrong.
   # Patches from perf-tools-next, should be in 6.11 or hopefully backported.
-  patches = lib.optionals (lib.versionAtLeast kernel.version "6.10") [
+  patches = lib.optionals (lib.versions.majorMinor kernel.version == "6.10") [
```
Comment (alyssais) [inline pkgs/os-specific/linux/kernel/perf/default.nix:66]: unrelated, but is there any reason we need to build perf for every kernel version?

Comment (K900) [reply pkgs/os-specific/linux/kernel/perf/default.nix:66]: Not really AFAIK.

**Comment by K900** [comment]
Tested: Hetzner VM (x86_64), Asus GA402 laptop (x86_64), Orange Pi 3B (aarch64, new addition), Orange Pi 5 (aarch64, with some patches), CM3588 NAS KIT (aarch64, with some patches).

File: pkgs/os-specific/linux/kernel/perf/default.nix (hunk 1 above)
Comment (phip1611) [reply pkgs/os-specific/linux/kernel/perf/default.nix:66]: You sure that this fixes perf? Fails for me @K900 @alyssais 

```
rror: builder for '/nix/store/7m0fhi4mdib79x91fadkq17d0xwvrmjs-perf-linux-6.11.drv' failed with exit code 1;
       last 17 log lines:
       > Running phase: unpackPhase
       > unpacking source archive /nix/store/xrhpizkr37hc6zj55ab2spsnl3gr8h64-linux-6.11.tar.xz
       > source root is linux-6.11
       > setting SOURCE_DATE_EPOCH to timestamp 1726412276 of file linux-6.11/virt/lib/irqbypass.c
       > Running phase: patchPhase
       > applying patch /nix/store/cklsdshnrdvfifqvh9v1wn403n18yds6-?id=0f0e1f44569061e3dc590cd0b8cb74d8fd53706b
       > patching file tools/build/feature/test-libtracefs.c
       > Reversed (or previously applied) patch detected!  Assume -R? [n]
       > Apply anyway? [n]
       > Skipping patch.
       > 1 out of 1 hunk ignored -- saving rejects to file tools/build/feature/test-libtracefs.c.rej
       > patching file tools/perf/Makefile.config
       > Hunk #1 FAILED at 183.
       > Hunk #2 FAILED at 1178.
       > Hunk #3 FAILED at 1194.
       > Hunk #4 FAILED at 1315.
       > 4 out of 4 hunks FAILED -- saving rejects to file tools/perf/Makefile.config.rej
```

Comment (K900) [reply pkgs/os-specific/linux/kernel/perf/default.nix:66]: Yes it does, I have it built right now on my system.

Comment (phip1611) [reply pkgs/os-specific/linux/kernel/perf/default.nix:66]: Hm. Any idea why it fails on my side?

I'm using `packages = [ config.boot.kernelPackages.perf ]`

Comment (phip1611) [reply pkgs/os-specific/linux/kernel/perf/default.nix:66]: Interesting. My failure only happens with `kernelPackages_6_11` from `nixos-24.05`

**Pull request merged**